lib/Dialect/Torch/Transforms/ScalarizeShapes.cpp

```diff
@@ -530,11 +530,139 @@ class FoldAtenUnsqueezePattern : public OpRewritePattern<AtenUnsqueezeOp> {
                                                      none, none, none, none);
       return success();
     }
+    auto squeezeOp = op.getSelf().getDefiningOp<AtenSqueezeDimOp>();
+    if (squeezeOp && resultTy.getSizes().size() == 1) {
```

Comment:
Shouldn't we include a check that the input / output rank are the same as well?

Reply:
If the result rank of unsqueeze is 1, then it can only have shape [1]. Therefore, if the input came from a squeeze.dim op, it must have been a squeeze from [1] to [].